invalid syntax on async keyword

When I flake8 the file from the command line everything passes, but I get invalid syntax from the plugin.

I'm using pyenv, but the shims seem to be loading correctly below, since it pulls the path from the global python version.

```
$ pyenv global
3.5.0
```

Output from editor open to file lint (conftest.py)

```
DPI scale: 1
startup, version: 3083 linux x64 channel: stable
executable: /opt/sublime_text/sublime_text
working dir: /
packages path: /home/numberoverzero/.config/sublime-text-3/Packages
state path: /home/numberoverzero/.config/sublime-text-3/Local
zip path: /opt/sublime_text/Packages
zip path: /home/numberoverzero/.config/sublime-text-3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.10723
startup time: 0.205469
first paint time: 0.255404


<SNIP> reloading plugin spam </SNIP>


reloading plugin SublimeLinter-flake8.linter
SublimeLinter: flake8 linter loaded 
reloading plugin SublimeLinter.commands
reloading plugin SublimeLinter.sublimelinter
plugins loaded
SublimeLinter: debug mode: on 
SublimeLinter: temp directory: /tmp/SublimeLinter3-numberoverzero 
SublimeLinter: find_python(version='3', script=None, module=None) 
SublimeLinter: find_python: requested version = {'minor': None, 'major': 3} 
SublimeLinter: user shell: /bin/bash 
SublimeLinter: computed PATH using /bin/bash:
/home/numberoverzero/.pyenv/shims
/home/numberoverzero/.pyenv/bin
/usr/local/sbin
/usr/local/bin
/usr/sbin
/usr/bin
/sbin
/bin
/usr/games
/usr/local/games
/usr/local/bin
/home/numberoverzero/bin
~/npm/bin
/home/numberoverzero/.rvm/bin
 
SublimeLinter: find_posix_python: python3 => /home/numberoverzero/.pyenv/shims/python3 
SublimeLinter: find_python: system python = /home/numberoverzero/.pyenv/shims/python3 
SublimeLinter: find_python: available version = {'minor': 5, 'major': 3} 
SublimeLinter: find_python: <= ('/home/numberoverzero/.pyenv/shims/python3', None, 3, 5) 
SublimeLinter: sys.path for /home/numberoverzero/.pyenv/shims/python3:
/home/numberoverzero/ws/bloop
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5/site-packages/arrow-0.7.0-py3.5.egg
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5/site-packages/jmespath-0.9.0-py3.5.egg
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5/site-packages/python_dateutil-2.4.2-py3.5.egg
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5/site-packages/six-1.10.0-py3.5.egg
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python35.zip
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5/plat-linux
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5/lib-dynload
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5/site-packages
 
SublimeLinter: flake8 imported <module 'flake8.engine' from '/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5/site-packages/flake8/engine.py'> 
Package Control: Skipping automatic upgrade, last run at 2016-01-12 19:34:04, next run at 2016-01-12 20:34:04 or after
SublimeLinter: flake8 version: 2.5.1 
SublimeLinter: flake8: (>= 2.2.2) satisfied by 2.5.1 
SublimeLinter: flake8 activated: <builtin> 
SublimeLinter: chdir not set or invalid, using /home/numberoverzero/ws/bottom/tests 
SublimeLinter: flake8: conftest.py <builtin> 
SublimeLinter: flake8 options: {'reporter': <class 'SublimeLinter-flake8.linter.Flake8.get_report.<locals>.Report'>, 'jobs': '1', 'max_complexity': -1} 
SublimeLinter: flake8 output:
conftest.py:22:14: E901 SyntaxError: invalid syntax 

```

I'm having the exact same problem. Here's my bug report:

Looking in the Sublime console SublimeLinter-flake8 loads the latest version of the underlying `flake8` library (`2.5.4` at the time of writing). Passing the file to flake8 in the console does not produce any error/warning output and exists with code 0. I checked the code from within the python 3 interpreter as well like this, giving me no error:
```
/usr/local/bin/python3
Python 3.5.0 (default, Oct 25 2015, 10:30:34)
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.72)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import flake8.run
>>> flake8.run.check_file('/path/to/app.py')
0
>>> flake8.__file__
'/usr/local/lib/python3.5/site-packages/flake8/__init__.py'
```

Here's the output that the Sublime console gives. As you can see from the file path below it's the same `flake8` package that I used programmatically above.
```
SublimeLinter: flake8 imported <module 'flake8.engine' from '/usr/local/lib/python3.5/site-packages/flake8/engine.py'> 
SublimeLinter: flake8 version: 2.5.4 
SublimeLinter: flake8: (>= 2.2.2) satisfied by 2.5.4 
SublimeLinter: flake8 activated: <builtin> 
SublimeLinter: flake8: __init__.py <builtin> 
SublimeLinter: flake8 options: {'reporter': <class 'SublimeLinter-flake8.linter.Flake8.get_report.<locals>.Report'>, 'max_complexity': -1, 'jobs': '1'} 
SublimeLinter: json activated: <builtin> 
reloading Packages/User/SublimeLinter.sublime-settings
SublimeLinter: flake8: __init__.py <builtin> 
SublimeLinter: flake8 options: {'reporter': <class 'SublimeLinter-flake8.linter.Flake8.get_report.<locals>.Report'>, 'max_complexity': -1, 'jobs': '1'} 
```

When I save the file I get:
```
SublimeLinter: flake8: app.py <builtin> 
SublimeLinter: flake8 options: {'reporter': <class 'SublimeLinter-flake8.linter.Flake8.get_report.<locals>.Report'>, 'max_complexity': -1, 'jobs': 'auto', 'ignore': ['E501', 'E711', 'E712']} 
SublimeLinter: flake8 output:
app.py:3:10: E901 SyntaxError: invalid syntax 
```

The file looks like this:
```python
import asyncio

async def hello_world():
    print("Hello World!")
...
```

--- a/linter.py
+++ b/linter.py
@@ -11,8 +11,7 @@
 
 """This module exports the Flake8 plugin linter class."""
 
-import os
-from SublimeLinter.lint import persist, PythonLinter
+from SublimeLinter.lint import PythonLinter
 
 
 class Flake8(PythonLinter):
@@ -56,83 +55,11 @@ class Flake8(PythonLinter):
         '--max-line-length=': None,
         '--max-complexity=': -1,
         '--jobs=': '1',
-        '--show-code=': False
+        'show-code': False,
+        'executable': ''
     }
-    inline_settings = ('max-line-length', 'max-complexity', 'show_code')
+    inline_settings = ('max-line-length', 'max-complexity')
     inline_overrides = ('select', 'ignore', 'builtins')
-    module = 'flake8.engine'
-    check_version = True
-
-    # Internal
-    report = None
-    show_code = False
-    pyflakes_checker_module = None
-    pyflakes_checker_class = None
-
-    @classmethod
-    def initialize(cls):
-        """Initialize the class after plugin load."""
-
-        super().initialize()
-
-        if cls.module is None:
-            return
-
-        # This is tricky. Unfortunately pyflakes chooses to store
-        # builtins in a class variable and union that with the builtins option
-        # on every execution. This results in the builtins never being removed.
-        # To fix that, we get a reference to the pyflakes.checker module and
-        # pyflakes.checker.Checker class used by flake8. We can then reset
-        # the Checker.builtIns class variable on each execution.
-
-        try:
-            from pkg_resources import iter_entry_points
-        except ImportError:
-            persist.printf('WARNING: {} could not import pkg_resources.iter_entry_points'.format(cls.name))
-        else:
-            for entry in iter_entry_points('flake8.extension'):
-                check = entry.load()
-
-                if check.name == 'pyflakes':
-                    from pyflakes import checker
-                    cls.pyflakes_checker_module = checker
-                    cls.pyflakes_checker_class = check
-                    break
-
-    def check(self, code, filename):
-        """Run flake8 on code and return the output."""
-
-        options = {
-            'reporter': self.get_report(),
-            'jobs': '1'   # No multiprocessing
-        }
-
-        type_map = {
-            'select': [],
-            'ignore': [],
-            'builtins': '',
-            'max-line-length': 0,
-            'max-complexity': 0,
-            'show-code': False
-        }
-
-        self.build_options(options, type_map, transform=lambda s: s.replace('-', '_'))
-        self.show_code = options.pop('show_code', False)
-
-        if persist.debug_mode():
-            persist.printf('{} options: {}'.format(self.name, options))
-
-        if self.pyflakes_checker_class is not None:
-            # Reset the builtins to the initial value used by pyflakes.
-            builtins = set(self.pyflakes_checker_module.builtin_vars).union(self.pyflakes_checker_module._MAGIC_GLOBALS)
-            self.pyflakes_checker_class.builtIns = builtins
-
-        linter = self.module.get_style_guide(**options)
-
-        return linter.input_file(
-            filename=os.path.basename(filename),
-            lines=code.splitlines(keepends=True)
-        )
 
     def split_match(self, match):
         """
@@ -148,34 +75,17 @@ class Flake8(PythonLinter):
         if near:
             col = None
 
-        if self.show_code:
+        if self.get_view_settings().get('show-code'):
             message = ' '.join([error or warning or '', message])
         return match, line, col, error, warning, message, near
 
-    def get_report(self):
-        """Return the Report class for use by flake8."""
-        if self.report is None:
-            from pep8 import StandardReport
+    def build_cmd(self, cmd=None):
+        """Return a tuple with the command line to execute."""
 
-            class Report(StandardReport):
-                """Provides a report in the form of a single multiline string, without printing."""
-
-                def get_file_results(self):
-                    """Collect and return the results for this file."""
-                    self._deferred_print.sort()
-                    results = ''
-
-                    for line_number, offset, code, text, doc in self._deferred_print:
-                        results += '{path}:{row}:{col}: {code} {text}\n'.format_map({
-                            'path': self.filename,
-                            'row': self.line_offset + line_number,
-                            'col': offset + 1,
-                            'code': code,
-                            'text': text
-                        })
-
-                    return results
-
-            self.__class__.report = Report
-
-        return self.report
+        executable = self.get_view_settings().get('executable', None)
+        if executable:
+            args = (cmd or self.cmd)[1:]
+            cmd = (executable, ) + args
+            return self.insert_args(cmd)
+        else:
+            return super().build_cmd(cmd)

